SCClassLibrary/Common/Control/Buffer.sc

```diff
@@ -11,38 +12,43 @@ Buffer {
 	*initClass { serverCaches = IdentityDictionary.new }
 
 	// doesn't send
-	*new { arg server, numFrames, numChannels, bufnum;
+	*new { arg server, numFrames, numChannels, sampleRate, bufnum;
 		server = server ? Server.default;
 		bufnum ?? { bufnum = server.nextBufferNumber(1) };
+		sampleRate ?? { sampleRate = server.sampleRate };
 		^super.newCopyArgs(
 			server,
 			bufnum,
 			numFrames,
-			numChannels
-		).sampleRate_(server.sampleRate).cache
+			numChannels,
+			sampleRate.asFloat
+		).cache
 	}
 
-	*alloc { arg server, numFrames, numChannels = 1, completionMessage, bufnum;
+	*alloc { arg server, numFrames, numChannels = 1, sampleRate, completionMessage, bufnum;
 		server = server ? Server.default;
 		bufnum ?? { bufnum = server.nextBufferNumber(1) };
+		sampleRate ?? { sampleRate = server.sampleRate };
 		^super.newCopyArgs(
 			server,
 			bufnum,
 			numFrames,
-			numChannels
-		).alloc(completionMessage).sampleRate_(server.sampleRate).cache
+			numChannels,
+			sampleRate.asFloat
+		).alloc(completionMessage).cache
 	}
 
-	*allocConsecutive { arg numBufs = 1, server, numFrames, numChannels = 1, completionMessage, bufnum;
+	*allocConsecutive { arg numBufs = 1, server, numFrames, numChannels = 1, sampleRate, completionMessage, bufnum;
 		var	bufBase;
 		bufBase = bufnum ?? { server.nextBufferNumber(numBufs) };
 		numFrames = numFrames.asInteger;
 		numChannels = numChannels.asInteger;
+		sampleRate ?? { sampleRate = server.sampleRate };
 		^Array.fill(numBufs, { |i|
 			// note, cannot use alloc or allocMsg here
 			// because those methods don't pass a buffer index for completion message
-			var newBuf = Buffer.new(server, numFrames, numChannels, i + bufBase);
-			server.sendMsg(\b_alloc, i + bufBase, numFrames, numChannels,
+			var newBuf = Buffer.new(server, numFrames, numChannels, sampleRate, i + bufBase);
+			server.sendMsg(\b_alloc, i + bufBase, numFrames, numChannels, sampleRate.asFloat,
```

Comment:
forward as is (float or nil).

IMPORTANT: new message arguments must be added at the end to ensure backwards compatibility!